Sources/CCFilterNode/CCFilterNode.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import Foundation
+
+private protocol CCFilterNodeDelegate: class {
+    func childNodeDidChangeSelection(_ childNode: CCFilterNode)
+}
+
+public class CCFilterNode {
+
+    // MARK: - Public properties
+
+    public let title: String
+    public let name: String
+    public var value: String?
+    public var numberOfResults: Int
+
+    public var isSelected: Bool {
+        didSet { delegate?.childNodeDidChangeSelection(self) }
+    }
+
+    // MARK: - Private properties
+
+    private(set) var children: [CCFilterNode] = []
+    private weak var delegate: CCFilterNodeDelegate?
+
+    // MARK: - Setup
+
+    public init(title: String, name: String, value: String? = nil, isSelected: Bool = false, numberOfResults: Int = 0) {
+        self.title = title
+        self.name = name
+        self.value = value
+        self.isSelected = isSelected
+        self.numberOfResults = numberOfResults
+    }
+
+    // MARK: - Public methods
+
+    func add(child: CCFilterNode, at index: Int? = nil) {
+        if let index = index { children.insert(child, at: index) }
```

Comment:
Not sure I like this `if else` formatting, I'd prefer:
```swift
if {
} else {
}
```

Reply:
:P I'll change it